src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/ValueListBuilder.Pop.cs

```diff
@@ -12,10 +12,10 @@ namespace System.Collections.Generic
     internal ref partial struct ValueListBuilder<T>
     {
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public T Pop()
+        public ref T Pop()
```

Comment:
This is pretty dangerous. Very easy to make a mistake by using the value after it has been overwritten.

Reply:
I'm not using the Pop function with the ref keyword, I just wanted to add the possibility to do so. I understand the concerns.

Reply:
If you do not actually needs this, this change should be undone.